ambry-protocol/src/main/java/com/github/ambry/protocol/DeleteRequest.java

```diff
@@ -147,7 +178,18 @@ static DeleteRequest readFrom(DataInputStream stream, ClusterMap map) throws IOE
       String clientId = Utils.readIntString(stream);
       BlobId id = new BlobId(stream, map);
       long deletionTimeInMs = stream.readLong();
-      return new DeleteRequest(correlationId, clientId, id, deletionTimeInMs);
+      return new DeleteRequest(correlationId, clientId, id, deletionTimeInMs, DELETE_REQUEST_VERSION_2, false);
+    }
+  }
+
+  private static class DeleteRequest_V3 {
+    static DeleteRequest readFrom(DataInputStream stream, ClusterMap map) throws IOException {
+      int correlationId = stream.readInt();
+      String clientId = Utils.readIntString(stream);
+      BlobId id = new BlobId(stream, map);
+      long deletionTimeInMs = stream.readLong();
+      boolean isForceDelete = stream.readByte() == 1;
+      return new DeleteRequest(correlationId, clientId, id, deletionTimeInMs, isForceDelete);
```

Comment:
Should add VERSION_3 here. This constructor would use version 2

Reply:
Done